docs/reference/method/MongoDBModelIndexInfo-getBucketSize.txt

```diff
@@ -0,0 +1,55 @@
+==========================================
+MongoDB\\Model\\IndexInfo::getBucketSize()
+==========================================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Definition
+----------
+
+.. phpmethod:: MongoDB\\Model\\IndexInfo::getBucketSize()
+
+   Return the bucket size of a :manual:`geoHaystack </core/geohaystack>` index. This method will return null
+   if the index is not a geoHaystack index.
+
+   .. code-block:: php
+
+      function getBucketSize(): integer|double|null
+
+Return Values
+-------------
+
+The bucket size of a geoHaystack index.
+
+Examples
+--------
+
+.. code-block:: php
+
+   <?php
+   $collection->createIndex(['x' => 'geoHaystack', 'y' => 1], ['bucketSize' => 5]);
+
+   $resultIterator = $collection->listIndexes();
+   $resultIterator->rewind();
+   $resultIterator->next();
+
+   var_dump($resultIterator->current()->getBucketSize());
```

Comment:
If we change this documentation to demonstrate `isGeoHaystack()`:

```php
$collection = (new MongoDB\Client)->selectCollection('test', 'places');

$collection->createIndex(['pos' => 'geoHaystack', 'x' => 1], ['bucketSize' => 5]);

foreach ($collection->listIndexes() as $index) {
    if ($index->isGeoHaystack()) {
        printf("%s has bucketSize: %d\n", $index->getName(), $index['bucketSize']);
    }
}
```

This should output:

> pos_geoHaystack_x_1 has bucketSize: 5